framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -1105,6 +1129,23 @@ public ClassOrInterfaceAnnos(@Nullable TypeDeclaration<?> javaParserNode) {
       classDeclaration = javaParserNode;
     }
 
+    // Makes a deep copy
+    @Override
+    public ClassOrInterfaceAnnos clone() {
+      try {
+        ClassOrInterfaceAnnos result = (ClassOrInterfaceAnnos) super.clone();
+        result.callableDeclarations = CollectionUtils.deepCopyValues(result.callableDeclarations);
+        result.fields = CollectionUtils.deepCopyValues(result.fields);
```

Comment:
I think using `clone` is error prone because it's easy to miss that a class hasn't properly implemented `clone`. For examile, this line (indirectly) calls FieldAnnos#clone, but FieldAnnos#clone hasn't been implemented yet. 